File: Royaltime.APIGateway/Entities/Product.cs
namespace APIGateway.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string Brand { get; set; }
        public string Type { get; set; }
        public string Gender { get; set; }
        public string Model { get; set; }
        public string DialColor { get; set; }
        public string StrapType { get; set; }
        public int QuantityInStock { get; set; }
        public string PublicId { get; set; }
        public ICollection<WishlistItem> Wishlist { get; set; }

    }
}

File: client/src/app/models/product.ts
export interface Product {
    id: number
    name: string
    description: string
    price: number
    pictureUrl: string
    brand: string
    type?: string
    gender?: string
    model?: string
    dialColor: string
    strapType: string
    quantityInStock?: number
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    brands: string[];
    genders: string[];
    models: string[];
    types: string[];
    pageNumber: number;
    pageSize: number;
  }